src/Psalm/Internal/Type/TypeParser.php

```diff
@@ -1566,14 +1560,19 @@ private static function extractKeyedIntersectionTypes(
             $keyed_intersection_types[self::extractIntersectionKey($callable_object_type)] = $callable_object_type;
         }
 
-        if ($any_object_type_found && $any_array_found) {
-            throw new TypeParseTreeException(
-                'Intersection types must be all objects or all keyed array.',
-            );
+        if ($any_object_type_found) {
+            if ($any_array_found) {
+                throw new TypeParseTreeException(
+                    'Intersection types must be all objects or all keyed array.',
+                );
+            }
+
+            if ($keyed_intersection_types === []) {
+                return ['object' => new TObjectWithProperties([], [], [], $object_from_docblock)];
```

Comment:
- [ ] Seems that this actually leads to the issuebuffer test issue.
will see how to improve that later this week